Automattic/jetpack
# Jetpack: optimize gutenberg initialization

## Proposed changes:
Optimize initialization of the `Jetpack_Gutenberg` class:
- Get rid of the `init()` method, use on-demand extension loading instead.
- Hardcode block directories instead of using expensive `glob()` function.

Performance improvements on my machine (Monorepo docker env, wp-admin "All Posts" page with 4 posts):
[`Jetpack_Gutenberg` initialization](https://github.com/Automattic/jetpack/pull/30875/files#diff-028fe0b6a09c6ed2b0a61f21d8db4d21cf8ca35c8904f94f78503619455fe31fL685-L687) before the changes: ~0.22s
With the changes applied: ~0.08s

### Other information:


## Jetpack product discussion
p1HpG7-lOi-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack
2. Go to "Settings" and enable VideoPress: `/wp-admin/admin.php?page=jetpack#/settings?term=videopress`
3. Edit or create a new post
4. Try using the "Calendly" block, make sure it shows up and works fine.
5. Try using the VideoPress block, confirm it's the new one (with the fancy icon, see screenshot), and it does NOT say `Enable Jetpack module to continue using VideoPress`
<img width="668" alt="Screen Shot 2023-05-24 at 17 39 04" src="https://github.com/Automattic/jetpack/assets/1341249/42c27e10-23be-41f2-953a-32d705d50241">
